rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,401 @@
+
+#### POST submission/{submission_id}/upload/file
+
+An authenticated user can upload a file directly from their local machine to a data set in their submission.
+
+Asset_uuid can be included to retrieve a new access token if you were unable to complete the upload before the token
+expired. An error will be returned if the upload has already been completed. The user must own the submission and be
+the one who initiated the upload to retrieve an additional access token.
+
+If the size of the file being upload is <5GB, then a pre-signed URL will be returned.
+If the size of the file is >5GB, then an access token will be returned.
+
+**Request:**
+
+| Parameter             | Description                                                                |
+| --------------------- | -------------------------------------------------------------------------- |
+| File size             | the size of the file to be uploaded                                        |
+| File Name             | the name of the file being uploaded                                        |
+| submission_id         | identifies the submission                                                  |
+| attestation           | A boolean value indicating if the file is for attestation                  |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used if resuming an existing upload |
+
+**Response:**
+
+| Key           | Description                                                   |
+| ------------- | ------------------------------------------------------------- |
+| Access Token  | allows the user to upload their file to S3 in multiple parts. |
+| Presigned URL | allows the user to upload their file to S3 in a single part.  |
+| dataset_id    | identifies the name of the new file created.                  |
+| status     | Provides the current status of the upload.			  |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
+| 400  | if the file type is invalid                                                                                            |
+
+#### Upload States
+
+When uploading a file it is important to know what state the upload is in. The API documentation returns a *status* in 
+response. This status informs the user the current state of the upload. An upload follows this state diagram:
+
+![Upload States](https://app.lucidchart.com/publicSegments/view/a952f3e9-2ef9-4398-866b-d1455feeaec5/image.png)
+Figure 2: Upload State Diagram
+
+The states are describes as follow:
+- Uploading: The file is actively being uploaded.
```

Comment:
Perhaps this is just a nice to have;  but we should have enough information to know how much data has been uploaded.  This information can be used to build a progress meter into the client.
I'm thinking about the case where a someones uploads 10GB over a 10MB/s link.   There would be a many minutes long delay where all they would know is "Uploading".